File: examples/javascript/function_bp_test.js
/**
 * Fixture for JavaScript function-breakpoint smoke tests (issue #295).
 * Runs as ESM (repo root package.json is "type": "module").
 * Line layout is asserted by tests/e2e/mcp-server-smoke-js-function-bp.test.ts:
 * compute is declared at line 9 (its entry statement is line 10).
 */
console.log('function bp fixture start');

function compute(x) {
  console.log('compute called with', x);
  return x * 2;
}

const sleep = (ms) => new Promise((r) => setTimeout(r, ms));

let helper = null;

(async () => {
  await sleep(300);
  compute(3);
  helper = await import('./function_bp_helper.js');
  compute(7);
  await sleep(200);
  helper.doWork();
  await sleep(3000);
  compute(99);
  console.log('FUNCTION_BP_FIXTURE_DONE');
})();

File: examples/java/FunctionBpTest.java
/**
 * Function-breakpoint e2e fixture (issue #292).
 * Line numbers are asserted by mcp-server-smoke-java-function-bp.test.ts —
 * keep the layout stable: the first statement of compute(int) is line 20,
 * of compute(String) line 25, of helper() line 30 (method entry locations
 * map to the first statement, not the declaration line).
 */
public class FunctionBpTest {
    public static void main(String[] args) throws InterruptedException {
        System.out.println("FunctionBpTest starting");
        Thread.sleep(3000); // window for setting breakpoints while running
        int a = compute(3);
        int b = compute(7);
        String s = compute("hi");
        String g = helper();
        System.out.println("Result: " + (a + b) + " " + s + " " + g);
    }

    static int compute(int x) {
        int doubled = x * 2;
        return doubled;
    }

    static String compute(String s) {
        String up = s.toUpperCase();
        return up;
    }

    static String helper() {
        String msg = "helped";
        return msg;
    }
}
